core-rust/core-api-server/src/core_api/conversions/addressing.rs

```diff
@@ -436,25 +430,33 @@ fn to_mapped_substate_id(substate_id: SubstateId) -> Result<MappedSubstateId, Ma
 
         SubstateId(RENodeId::Vault(_), _, offset) => {
             let substate_type_key = match offset {
-                SubstateOffset::ComponentTypeInfo(offset) => match offset {
-                    ComponentTypeInfoOffset::TypeInfo => {
-                        (SubstateType::ComponentInfo, SubstateKeyType::ComponentInfo)
-                    }
+                SubstateOffset::TypeInfo(offset) => match offset {
+                    TypeInfoOffset::TypeInfo => (SubstateType::TypeInfo, SubstateKeyType::TypeInfo),
                 },
                 SubstateOffset::Vault(offset) => match offset {
-                    VaultOffset::Vault => (SubstateType::Vault, SubstateKeyType::Vault),
+                    VaultOffset::Info => (SubstateType::VaultInfo, SubstateKeyType::VaultInfo),
+                    VaultOffset::LiquidNonFungible | VaultOffset::LockedNonFungible => (
+                        SubstateType::VaultNonFungible,
+                        SubstateKeyType::VaultNonFungible,
+                    ),
+                    VaultOffset::LiquidFungible | VaultOffset::LockedFungible => {
+                        (SubstateType::VaultFungible, SubstateKeyType::VaultFungible)
+                    }
```

Comment:
Here is another one of my guesses.

The same test (`TransactionStreamTest.test_core_api_can_submit_and_commit_transaction()`) demanded that a mapping for `Locked*` vaults exists as well.

⚠️ @talekhinezh - I explicitly overwritten your previous adjustments here (kinda blindly, just to satisfy the test), please review.

Reply:
Close, I've just pushed a fix